src/Microsoft.Health.Fhir.Shared.Api/Controllers/OperationDefinitionController.cs

```diff
@@ -0,0 +1,137 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Net;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Api.Configs;
+using Microsoft.Health.Fhir.Api.Features.ActionResults;
+using Microsoft.Health.Fhir.Api.Features.Filters;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Messages.Operation;
+using Microsoft.Health.Fhir.Shared.Core.Extensions;
+
+namespace Microsoft.Health.Fhir.Api.Controllers
+{
+    /// <summary>
+    /// Controller that will handle all requests for OperationDefinition of
+    /// operations that are supported by our fhir-server and that do not have
+    /// an explicit definition in the HL7 website.
+    /// </summary>
+    [ServiceFilter(typeof(OperationOutcomeExceptionFilterAttribute))]
+    public class OperationDefinitionController : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly OperationsConfiguration _operationConfiguration;
+        private readonly FeatureConfiguration _featureConfiguration;
+
+        public OperationDefinitionController(
+            IMediator mediator,
+            IOptions<OperationsConfiguration> operationsConfig,
+            IOptions<FeatureConfiguration> featureConfig)
+        {
+            EnsureArg.IsNotNull(mediator, nameof(mediator));
+            EnsureArg.IsNotNull(operationsConfig?.Value, nameof(operationsConfig));
+            EnsureArg.IsNotNull(featureConfig?.Value, nameof(featureConfig));
+
+            _mediator = mediator;
+            _operationConfiguration = operationsConfig.Value;
+            _featureConfiguration = featureConfig.Value;
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.ReindexOperationDefinition, Name = RouteNames.ReindexOperationDefintion)]
+        [AllowAnonymous]
+        public async Task<IActionResult> ReindexOperationDefintion()
+        {
+            return await GetOperationDefinitionAsync(OperationsConstants.Reindex);
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.ResourceReindexOperationDefinition, Name = RouteNames.ResourceReindexOperationDefinition)]
+        [AllowAnonymous]
+        public async Task<IActionResult> ResourceReindexOperationDefinition()
```

Comment:
One final thought, is there a way we could serve these by-convention, i.e. without needing to code in each one?

Reply:
I tried binding multiple routes to the same method but that doesn't seem to work out. I am open to suggestions on how to avoid coding each one separately